src/node/rpc/frontend.h

```diff
@@ -258,20 +258,36 @@ namespace ccf
         if (!endpoint->authn_policies.empty())
         {
           std::string auth_error_reason;
+          std::vector<ccf::ODataErrorDetails> errordetails;
           for (const auto& policy : endpoint->authn_policies)
           {
             identity = policy->authenticate(tx, ctx, auth_error_reason);
             if (identity != nullptr)
             {
               break;
             }
+            else
+            {
+              // Collate error details
+              errordetails.push_back(
+                {policy->SECURITY_SCHEME_NAME,
+                 ccf::errors::InvalidAuthenticationInfo,
+                 auth_error_reason});
+            }
           }
 
           if (identity == nullptr)
           {
-            // If none were accepted, let the last set an error
+            // If none were accepted, let the last set the response header
             endpoint->authn_policies.back()->set_unauthenticated_error(
```

Comment:
I think we should go to each of the implementations of `set_unauthenticated_error`, and remove the calls to `ctx->set_error`. They construct an OData object for their own error and insert that into the body, but that's then overwritten by the call to `ctx->set_error` below. We should remove those calls that get overwritten to avoid confusion.